pkg/api/vlabs/validate.go

```diff
@@ -274,14 +277,21 @@ func (a *Properties) Validate() error {
 	return nil
 }
 
-func (a *Properties) ValidateNetworkPolicy() error {
-
+func (a *Properties) validateNetworkPolicy() error {
 	if a.OrchestratorProfile.OrchestratorType != Kubernetes {
 		return nil
 	}
 
 	networkPolicy := a.OrchestratorProfile.KubernetesConfig.NetworkPolicy
-	if networkPolicy != "" && networkPolicy != "none" && networkPolicy != "calico" {
```

Comment:
This might fall through if a new network policy is added. Can you change this to: if networkPolicy == "azure"